File: DS_A/C++/math/eratosthenes_sieve.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

vector<int> eratasfen(int limit) {
    vector<bool> is_p(limit + 1, true);
    is_p[0] = is_p[1] = false;

    for (int i = 2; i <= sqrt(limit); ++i) {
        if (is_p[i]) {
            for (int j = i * i; j <= limit; j += i) {
                is_p[j] = false;
            }
        }
    }

    vector<int> pr;
    for (int i = 2; i <= limit; ++i) {
        if (is_p[i]) {
            pr.push_back(i);
        }
    }

    return pr;
}

File: DS_A/Python/math/eratosthenes_sieve.py
def eratasfen(limit):
	is_p = [True] * (limit+1)
	is_p[0] = is_p[1] = False
	
	for i in range(1,int(limit**0.5)+1):
		if is_p[i]:
			for j in range(i*i,limit+1,i):
				is_p[j]=False
	
	pr = [i for i in range(2,limit+1) if is_p[i]]
	return is_p,pr
